packages/braid-design-system/src/lib/components/private/Popover/Popover.tsx

```diff
@@ -0,0 +1,342 @@
+  return Math.min(Math.max(preferred, min), max);
+}
+
+export const Popover = forwardRef<HTMLElement, PopoverProps>(
+  (
+    {
+      id,
+      role,
+      align: alignProp = 'left',
+      width = 'content',
+      placement = 'bottom',
+      lockPlacement = false,
+      offsetSpace = 'none',
+      open,
+      onClose,
+      triggerRef,
+      enterFocusRef,
+      delayVisibility = false,
+      children,
+    },
+    forwardedRef,
+  ) => {
+    const [triggerPosition, setTriggerPosition] = useState<
+      Position | undefined
+    >(undefined);
+
+    const ref = useRef<HTMLElement>(null);
+    useImperativeHandle(forwardedRef, () => ref.current as HTMLElement);
+
+    const [horizontalOffset, setHorizontalOffset] = useState(0);
+    const [flipPlacement, setFlipPlacement] = useState<Placement>(placement);
+
+    const showPopover = open && triggerPosition;
+
+    const align = alignProp === 'center' ? 'left' : alignProp;
+
+    const handleOnClose = useCallback(
+      ({ closeTrigger }: { closeTrigger: 'backdrop' | 'keyboard' }) => {
+        if (closeTrigger !== 'backdrop' && triggerRef) {
+          if (triggerRef?.current) {
+            triggerRef.current.focus();
+          } else {
+            // eslint-disable-next-line no-console
+            console.error(
+              dedent`
+          The triggerRef element could not be found.
+          Ensure it is being assigned to the Popover trigger, and is available on close.
+          `,
+            );
+          }
+        }
+
+        if (onClose) {
+          onClose();
+        }
+      },
+      [onClose, triggerRef],
+    );
+
+    useEffect(() => {
+      if (!open || !onClose) {
+        return;
+      }
+
+      const handleKeydown = (event: globalThis.KeyboardEvent) => {
+        if (event.key === 'Escape' || event.key === 'Tab') {
+          handleOnClose({ closeTrigger: 'keyboard' });
+        }
+      };
+
+      window.addEventListener('keydown', handleKeydown);
+
+      return () => {
+        window.removeEventListener('keydown', handleKeydown);
+      };
+    }, [open, onClose, handleOnClose]);
+
+    useEffect(() => {
+      if (!open) {
+        setTriggerPosition(undefined);
+        setHorizontalOffset(0);
+        return;
+      }
+
+      setTriggerPosition(getPosition(triggerRef.current));
+      // Without timeout, focus will not work on first render
+      // Needs to be 10ms to work in Safari - 0 for other browsers
+      setTimeout(() => {
+        if (!enterFocusRef) {
+          return;
+        }
+
+        if (enterFocusRef.current) {
+          enterFocusRef.current.focus();
+        } else {
+          // eslint-disable-next-line no-console
+          console.error(
+            dedent`
+                The enterFocusRef element could not be found.
+                Ensure it is being assigned to a child element of Popover, and is available on open.
+                `,
+          );
```

Comment:
This will log errors in production. I would suggest wrapping in dev warning block

```ts
enterFocusRef.current?.focus();

if (process.env.NODE_ENV !== 'production') {
  if (!enterFocusRef.current) {
    console.error(
      dedent`
        The enterFocusRef element could not be found.
        Ensure it is being assigned to a child element of Popover, and is available on open.
      `,
    );
  }
}
```